packages/cli/src/util/link/ensure-link.ts

```diff
@@ -38,6 +38,7 @@ export async function ensureLink(
     (link.status === 'linked' && opts.forceDelete) ||
     link.status === 'not_linked'
   ) {
+    opts.link = link;
     link = await setupAndLink(client, cwd, opts);
```

Comment:
**question (non-blocking):** What is this change doing, exactly?

It seems that `opts.link` is populated so that `setupAndLink` can later not call `getLinkedProject`. In this case (from `ensure-link.ts`), that means there's no change because in this condition `link` will always be truthy.

In the other place where `setupAndLink` is called (from `commands/dev/dev.ts`), `link` is not passed in, which will cause `getLinkedProject to be called. Should that invocation be updated too?

If so, then all invocations of `setupAndLink` would be called with a truthy `link`, making the change in `setup-and-link.ts` effectively a no-op.

What's the intent behind this change?

Reply:
What I found was this error https://github.com/vercel/vercel/blob/49f453742b190c989447bedf3c85947287086688/packages/cli/src/util/projects/link.ts#L130-L136 was being rendered twice because `getLinkedProject()` was getting called twice. I figured, if we already fetched the project, no sense doing it twice, so I pass the `link` through.

I did notice `commands/dev/dev.ts` calls `setupAndLink()`, but I missed that `getLinkedProject()` is called right before.

I believe we should make `link` a required property of `SetupAndLinkOptions`, pass `link` into `setupAndLink()` in `commands/dev/dev.ts`, and remove the `getLinkedProject()` call in `setup-and-link.ts`.